custom_components/candy/client/__init__.py

```diff
@@ -22,31 +30,55 @@
 
 
 class CandyClient:
-
-    def __init__(self, session: ClientSession, device_ip: str, encryption_key: str, use_encryption: bool):
-        self.session = session  # Session is the default HA session, shouldn't be cleaned up
+    def __init__(
+        self,
+        session: ClientSession,
+        device_ip: str,
+        encryption_key: str,
+        use_encryption: bool,
+    ):
+        self.session = (
+            session  # Session is the default HA session, shouldn't be cleaned up
+        )
         self.device_ip = device_ip
         self.encryption_key = encryption_key
         self.use_encryption = use_encryption
 
-    @backoff.on_exception(backoff.expo, aiohttp.ClientError, max_tries=10, logger=__name__)
+    @backoff.on_exception(
+        backoff.expo, aiohttp.ClientError, max_tries=10, logger=__name__
+    )
     @backoff.on_exception(backoff.expo, TimeoutError, max_tries=10, logger=__name__)
-    async def status_with_retry(self) -> Union[WashingMachineStatus, TumbleDryerStatus, DishwasherStatus, OvenStatus]:
+    async def status_with_retry(
+        self,
+    ) -> Union[
+        WashingMachineStatus, TumbleDryerStatus, DishwasherStatus, OvenStatus, HobStatus
+    ]:
         return await self.status()
 
-    async def status(self) -> Union[WashingMachineStatus, TumbleDryerStatus, DishwasherStatus, OvenStatus]:
+    async def status(
+        self,
+    ) -> Union[
+        WashingMachineStatus, TumbleDryerStatus, DishwasherStatus, OvenStatus, HobStatus
+    ]:
         url = _status_url(self.device_ip, self.use_encryption)
         async with _LIMITER, self.session.get(url) as resp:
             if self.use_encryption:
-                resp_hex = await resp.text()  # Response is hex encoded, either encrypted or not
+                resp_hex = (
+                    await resp.text()
+                )  # Response is hex encoded, either encrypted or not
                 if self.encryption_key != "":
-                    decrypted_text = decrypt(self.encryption_key.encode(), bytes.fromhex(resp_hex))
+                    decrypted_text = decrypt(
+                        self.encryption_key.encode(), bytes.fromhex(resp_hex)
+                    )
                 else:
                     # Response is just hex encoded without encryption (details in detect_encryption())
                     decrypted_text = bytes.fromhex(resp_hex)
                 resp_json = json.loads(decrypted_text)
             else:
-                resp_json = await resp.json(content_type="text/html")
+                try:
+                    resp_json = await resp.json(content_type="text/html")
+                except Exception:
```

Comment:
What's the reason for adding this retry?

Reply:
This is because some appliance have content type text/html and not application/json. 
This code trw first with explicit conten type and then if is exception we try to get respone without explicit content type.
I think that i had problems with hub.

Reply:
Yes, I know that respond with a content type of `text/html`, this is already handled by the current code. I don't see the point of retrying the request with another content type.

Reply:
Then we skip it? I can skip iw you allready check it